runtime/autoload/health/provider.vim

```diff
@@ -703,6 +703,13 @@ function! s:check_perl() abort
     return
   endif
 
+  call s:system(['perl', '-e', 'use v5.20'])
+  if v:shell_error
```

Comment:
I meant that you should update the warning above instead of adding another one.

Reply:
Ok, I've reworked this a bit more. I've made an additional change here to improve the health check. We were not honouring the ```g:perl_host_prog``` variable when we were doing the health checks, instead we were just running ```perl```.